tree/treeplayer/test/regressions.cxx

```diff
@@ -173,3 +175,20 @@ TEST(TTreeReaderRegressions, IndexedFriend)
 
    gSystem->Unlink(fname);
 }
+
+// ROOT-8842
+TEST(TTreeReaderRegressions, ValueFastTuple)
+{
+    TNtuple tree("tuple", "ROOT-8842", "px:py:pz:energy");
+    for(auto i=0; i<1000000; ++i)
+        tree.Fill(i,i+1,i+2,i+3);
+    ROOT::Experimental::TTreeReaderFast reader(&tree);
+    ROOT::Experimental::TTreeReaderValueFast<float> px(reader, "px");
+    ROOT::Experimental::TTreeReaderValueFast<float> py(reader, "py");
+    ROOT::Experimental::TTreeReaderValueFast<float> pz(reader, "pz");
+    double total = 0.0;
+    // reader.SetEntry(0); If I uncomment this, the crash disappears
```

Comment:
Why/How is this call fixing the problem?

Reply:
I have no idea :s

Reply:
We would need to trace that down to make sure that there is not some other negative side-effects that is not being tickled in this short test.